nx-X11/programs/Xserver/dix/dispatch.c

```diff
@@ -3647,17 +3634,11 @@ void InitClient(ClientPtr client, int i, void * ospriv)
     if (i)
     {
 	client->closeDownMode = DestroyAll;
```

Comment:
Like I said, in the same way upstream did.

Replace the `if (i) { ... } else { ... }` section with one single `client->closeDownMode = i ? DestroyAll : RetainPermanent;` statement.